src/x25519.rs

```diff
@@ -199,15 +197,13 @@ impl<'a> From<&'a ReusableSecret> for PublicKey {
 #[cfg_attr(feature = "zeroize", derive(Zeroize))]
 #[cfg_attr(feature = "zeroize", zeroize(drop))]
 #[derive(Clone)]
-pub struct StaticSecret(
-    #[cfg_attr(feature = "serde", serde(with = "AllowUnreducedScalarBytes"))] pub(crate) Scalar,
-);
+pub struct StaticSecret([u8; 32]);
```

Comment:
We have to be careful here. It would be really bad if this was a breaking change that made it impossible to deserialize static secrets which were made in version 1.x. Fortunately I don't believe this is the case. The `serde::Serialize` [impl for `Scalar`](https://github.com/dalek-cryptography/curve25519-dalek/blob/99c0520aa79401b69fb51d38172cd58c6a256cfb/src/scalar.rs#L441-L455) appears to be identical to what's done for `[u8; 32]`. I manually verified this with `bincode::serialize` as an example format.

I want to be perfectly sure of this, though. @tarcieri can you comment?

Reply:
Every 256-bit string is a valid X25519 private key and that property definitely needs to be preserved

Reply:
Are you also convinced that the above change doesn't change the wire representation?

Reply:
I can give it a utACK. It's certainly more straightforward.